File: code/graphs/wgraph-adt.py
class WGraph:
    def __init__(self, n):
        self.V = n                          # number of vertices
        self.adj = [[] for _ in range(n)]   # adj[u] = list of (v, w) pairs

    def add_edge(self, u, v, w):
        self.adj[u].append((v, w))
        # self.adj[v].append((u, w))        # add this line for an UNDIRECTED edge

File: code/graphs/wgraph-adt.rs
struct WGraph {
    adj: Vec<Vec<(usize, i64)>>,        // adj[u] = list of (v, weight) pairs
}
impl WGraph {
    fn new(n: usize) -> Self {
        WGraph { adj: vec![Vec::new(); n] }
    }
    fn add_edge(&mut self, u: usize, v: usize, w: i64) {
        self.adj[u].push((v, w));
        // self.adj[v].push((u, w));      // add this line for an UNDIRECTED edge
    }
}
